scripts/scripts-1.js

```diff
@@ -0,0 +1,171 @@
+function getMinNumberFromParams(numb_1, numb_2, numb_3, numb_4) {
+
+  const minimalNumber = Math.min(numb_1, numb_2, numb_3, numb_4);
+
+  if (numb_1 === undefined || numb_2 === undefined || numb_3 === undefined || numb_4 === undefined) {
+
+    return 'Undefined value!';
+
+  } else if (numb_1 === '' || numb_2 === '' || numb_3 === '' || numb_4 === '') {
+
+    return 'Your value is empty';
+
+  } else if (isNaN(minimalNumber)) {
+
+    return 'Your value is not a number!';
+
+  }
+
+  else
+
+    return minimalNumber;
+
+};
+
+function greatestСommonDivisor(numb_1, numb_2) {
+
+  const remainder = numb_1 % numb_2;
+
+  if (numb_2 === undefined || numb_1 === undefined) {
+
+    return 'Undefined value!';
+
+  } else if (isNaN(numb_2) || isNaN(numb_1)) {
+
+    return 'Your value is not a number!';
+
+  } else if (numb_2 !== 0) {
+
+    return greatestСommonDivisor(numb_2, remainder);
+  }
+
+  return numb_1;
+};
+
+function perfectNumber(numb) {
+
+  let sum = 0;
+
+  for (i = 1; i <= numb / 2; i++) {
+
+    if (numb % i === 0) {
+
+      sum += i;
+
+    }
+
+  }
+
+  if (sum === numb && sum !== 0) {
+
+    return 'Your number is perfect';
+
+  } else if (numb == null) {
+
+    return 'Your value is empty';
+
+  } else if (numb === undefined || numb === "") {
+
+    return 'Undefined value!';
+
+  } else if (isNaN(numb)) {
+
+    return 'Your value is not a number!';
+
+  } else if (numb < 0) {
+
+    return 'Wrong! Negative value';
+
+  }
+
+  else {
+
+    return 'Your number is not perfect';
+
+  }
+
+};
+
+function sumFromOneNumberToTwo(numb_1, numb_2) {
```

Comment:
Here the same issue with validation/calculation and with strings validation
<img width="566" alt="image" src="https://user-images.githubusercontent.com/84010501/191557640-df8ed2be-4824-4a05-ab80-cbeaf4aa59ba.png">